src/org/ods/OdsContext.groovy

```diff
@@ -334,13 +382,23 @@ class OdsContext implements Context {
   }
 
   private String retrieveGitBranch() {
-    def pipelinePrefix = "${config.openshiftProjectId}/${config.openshiftProjectId}-"
-    def buildConfigName = config.jobName.substring(pipelinePrefix.size())
-
-    script.sh(
-      returnStdout: true,
-      script: "oc get bc/${buildConfigName} -n ${config.openshiftProjectId} -o jsonpath='{.spec.source.git.ref}'"
-    ).trim()
+    def branch
+    if (this.getLocalCheckoutEnabled()) {
+      def pipelinePrefix = "${config.openshiftProjectId}/${config.openshiftProjectId}-"
+      def buildConfigName = config.jobName.substring(pipelinePrefix.size())
+
+      branch = script.sh(
+              returnStdout: true,
+              script: "oc get bc/${buildConfigName} -n ${config.openshiftProjectId} -o jsonpath='{.spec.source.git.ref}'"
+      ).trim()
+    } else {
+      // in case code is already checked out, OpenShift build config can not be used for retrieving branch
+      branch = script.sh(
+              returnStdout: true,
+              script: "git rev-parse --abbrev-ref HEAD"
+      ).trim()
```

Comment:
I'm not entirely sure this will always be correct. Could it be passed from the MRO pipeline?

Reply:
Based on the assumption that we only checkout branches below `refs/heads/` (see below) all my tests on Jenkins succeeded.
https://github.com/opendevstack/ods-jenkins-shared-library/blob/aa765fc7789358d1470000202a6e26b1650c401d/src/org/ods/OdsPipeline.groovy#L192

If it should be possible to build /refs/tags, SHA, etc., my approach doesn't work. Then we also have to change the checkout.